src/IonEventStream.ts

```diff
@@ -0,0 +1,410 @@
+                            writer.writeDecimal(tempEvent.ionValue, tempEvent.annotations);
+                            break;
+                        case IonTypes.FLOAT :
+                            writer.writeFloat32(tempEvent.ionValue, tempEvent.annotations);
+                            break;
+                        case IonTypes.NULL :
+                            writer.writeNull(tempEvent.ionType.bid, tempEvent.annotations);
+                            break;
+                        case IonTypes.TIMESTAMP :
+                            break;
+                        case IonTypes.CLOB :
+                            tempBuf = [];
+                            for(let i = 0; i < tempEvent.ionValue.length; i++){
+                                tempBuf.push(tempEvent.ionValue.charCodeAt(i))
+                            }
+                            writer.writeClob(tempBuf, tempEvent.annotations);
+                            break;
+                        case IonTypes.BLOB :
+                            tempBuf = [];
+                            for(let i = 0; i < tempEvent.ionValue.length; i++){
+                                tempBuf.push(tempEvent.ionValue.charCodeAt(i))
+                            }
+                            writer.writeBlob(tempBuf, tempEvent.annotations);
+                            break;
+                        default :
+                            throw new Error("unexpected type: " + tempEvent.ionType.name);
+                    }
+                    break;
+                case IonEventType.CONTAINER_START:
+                    switch(tempEvent.ionType) {
+                        case IonTypes.STRUCT :
+                            writer.writeStruct(tempEvent.annotations, false);//need to change.
+                            break;
+                        case IonTypes.LIST :
+                            writer.writeList(tempEvent.annotations, false);
+                            break;
+                        case IonTypes.SEXP :
+                            writer.writeSexp(tempEvent.annotations, false);
+                            break;
+                    }
+
+                case IonEventType.CONTAINER_END:
+                    writer.endContainer();
+                    break;
+                case IonEventType.STREAM_END:
+                    writer.close();
+                    break;
+                case IonEventType.SYMBOL_TABLE:
+                    throw new Error("symboltables unsupported.");
+                default:
+                    throw new Error("Unexpected event type: "+ tempEvent.eventType);
+
+            }
+        }
+    }
+
+    getEvents() : IonEvent[]{
+        return this.eventStream;
+    }
+
+    equals(expected : IonEventStream) {
+        let actualIndex : number = 0;
+        let expectedIndex : number = 0;
+        for(; actualIndex < this.eventStream.length && expectedIndex < expected.eventStream.length; actualIndex++, expectedIndex++) {
```

Comment:
What if the streams differ in length? Remember, streams of different lengths may still be equivalent (in the case when they have a different number of SYMBOL_TABLE events, since SYMBOL_TABLE events aren't considered for equality).

Reply:
nvm gonna while loop